plugins/modules/ndo_physical_interface.py

```diff
@@ -0,0 +1,376 @@
+    interfaces = module.params.get("interfaces")
+    if interfaces:
+        interfaces = ",".join(interfaces)
+    physical_interface_type = module.params.get("physical_interface_type")
+    physical_policy_uuid = module.params.get("physical_policy_uuid")
+    physical_policy = module.params.get("physical_policy")
+    breakout_mode = module.params.get("breakout_mode")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+    object_description = "Physical Interface Profile"
+    path = "/fabricResourceTemplate/template/interfaceProfiles"
+
+    existing_physical_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("interfaceProfiles", [])
+
+    if name or uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_physical_interfaces,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_physical_interfaces
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        if physical_policy and not physical_policy_uuid:
+            fabric_policy_template = MSOTemplate(mso, "fabric_policy", physical_policy.get("template"))
+            fabric_policy_template.validate_template("fabricPolicy")
+            physical_policy_uuid = fabric_policy_template.get_interface_policy_group_uuid(physical_policy.get("name"))
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+            mso_values_remove = list()
+
+        mso_values = dict(
+            name=name,
+            description=description,
+            nodes=nodes,
+            interfaces=interfaces,
+            policyGroupType=physical_interface_type,
+        )
+
+        if physical_interface_type == "physical" and physical_policy_uuid:
+            mso_values["policy"] = physical_policy_uuid
+
+        if physical_interface_type == "breakout" and breakout_mode:
+            mso_values["breakoutMode"] = breakout_mode
+
+        if interface_descriptions:
+            mso_values["interfaceDescriptions"] = format_interface_descriptions(interface_descriptions, "")
+
+        if match:
+
+            if physical_interface_type and match.details.get("policyGroupType") != physical_interface_type:
+                mso.fail_json(msg="ERROR: Physical Interface type cannot be changed.")
+
+            update_path = "{0}/{1}".format(path, match.index)
+
+            if interface_descriptions == [] and proposed_payload.get("interfaceDescriptions"):
```

Comment:
Please add a documentation note to explain the update and clear processes of the "interface_descriptions".